### PR TITLE
Fix load theme sections

### PR DESCRIPTION
**Pull request by jonathanmoore**
Fixes a Theme Editor issue where JS errors were triggered and Section JS failed to reload after modifying section setting, adding a block, reordering, or any other event that triggers `shopify:section:load`.

- Fixed null container and type values with a new method to get the section type inside the Theme Editor #34
- Added a new `_onLoad` method to trigger the `onLoad` of Section js when the `shopify:section:load` event is fired #37 

_This approach should replace the open PR #36_

### CODE REVIEW

File: packages/theme-sections/theme-sections.js [hunk 1]
```diff
@@ -212,13 +212,16 @@ function normalizeContainers(containers) {
...
-    var id = event.detail.sectionId;
-    var container = event.target.querySelector(
-      '[' + SECTION_ID_ATTR + '="' + id + '}"]'
+  document.addEventListener('shopify:section:load', event => {
+    const container = event.target;
+    const containerDataset = container.dataset;
+
+    // Hack solution - pulling the first data-attribute since the identifier is unknown
+    const sectionData = JSON.parse(
+      containerDataset[Object.keys(containerDataset)[0]]
```
Comment (jonathanmoore) [inline packages/theme-sections/theme-sections.js]: Currently this code makes the assumption that this specific data-attribute will always be returned first. There has to be a better way to target the attribute to get the type value. Any ideas?

File: packages/theme-sections/theme-sections.js [hunk 2]
```diff
@@ -212,13 +212,16 @@ function normalizeContainers(containers) {
 }
 
 if (window.Shopify.designMode) {
-  document.addEventListener('shopify:section:load', function(event) {
-    var id = event.detail.sectionId;
-    var container = event.target.querySelector(
-      '[' + SECTION_ID_ATTR + '="' + id + '}"]'
+  document.addEventListener('shopify:section:load', event => {
```
Comment (t-kelly) [inline packages/theme-sections/theme-sections.js]: Avoiding ES6 syntax to support IE11 without using Babel which keeps filesize to a minimum. Please use `function() { ... }` not arrow functions.

File: packages/theme-sections/theme-sections.js [hunk 3]
```diff
@@ -212,13 +212,16 @@ function normalizeContainers(containers) {
...
+    const container = event.target;
+    const containerDataset = container.dataset;
+
+    // Hack solution - pulling the first data-attribute since the identifier is unknown
+    const sectionData = JSON.parse(
```
Comment (t-kelly) [inline packages/theme-sections/theme-sections.js]: Changing the way we access data attributes should be in a separate PR as it's a different problem than the bug of reloading the section. #36 is a more direct fix.

File: packages/theme-sections/section.js [hunk 4]
```diff
@@ -62,6 +64,10 @@ Section.prototype = {
     }
   },
 
+  _onLoad: function _onLoad(event) {
```
Comment (t-kelly) [inline packages/theme-sections/section.js]: I didn't include an `_onLoad()` method because it already exists as the `Section` constructor. `.onLoad()` is called on line 24.

**Comment by t-kelly** [comment]
@jonathanmoore `data-theme-editor-section-36545666210` is only intended for internal use for the editor and should not be relied on for external scripts. The editor team does not guarantee it won't change.

I recommend we still for the hard-coded data-attributes.

**Comment by jonathanmoore** [comment]
@t-kelly Going through and testing a bit more it does seem like typo PR did the trick.  The first pass of testing the typo PR it didn't seem to work because of issues I had in the theme's code.
https://github.com/Shopify/theme-scripts/pull/36
